source/common/config/filter_json.cc

```diff
@@ -0,0 +1,202 @@
+#include "common/config/filter_json.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+#include "common/config/json_utility.h"
+#include "common/config/protocol_json.h"
+#include "common/config/rds_json.h"
+#include "common/config/utility.h"
+#include "common/json/config_schemas.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Config {
+
+namespace {
+
+void translateComparisonFilter(const Json::Object& config,
+                               envoy::api::v2::filter::ComparisonFilter& filter) {
+  const std::string& op = config.getString("op");
```

Comment:
As an aside, it's very surprising to me that compiler and/or sanitizers don't catch this. I think must be getting lucky with how compiler uses temporary.